src/cargo/util/config.rs

```diff
@@ -190,16 +193,39 @@ impl Config {
             .map(AsRef::as_ref)
     }
 
+    /// Sets the clippy override. If this is true, clippy-driver is invoked instead of rustc.
+    pub fn set_clippy_override(&mut self, val: bool) {
+        self.clippy_override = val;
+    }
+
     /// Gets the path to the `rustc` executable.
     pub fn rustc(&self, ws: Option<&Workspace<'_>>) -> CargoResult<Rustc> {
         let cache_location = ws.map(|ws| {
             ws.target_dir()
                 .join(".rustc_info.json")
                 .into_path_unlocked()
         });
+        let wrapper = if self.clippy_override {
+            let tool = self.get_tool("clippy-driver")?;
+            let tool = paths::resolve_executable(&tool).map_err(|e| {
+                let rustup_in_path = self
+                    .get_tool("rustup")
+                    .and_then(|tool| paths::resolve_executable(&tool))
+                    .is_ok();
+                let has_rustup_env = std::env::var("RUSTUP_TOOLCHAIN").is_ok();
+                if rustup_in_path || has_rustup_env {
+                    failure::format_err!("{}: please run `rustup component add clippy`", e)
+                } else {
+                    failure::format_err!("{}: please install clippy component", e)
```

Comment:
I didn't know what to put here for the error message...

Reply:
"missing clippy-driver binary, please install clippy" should be ok

Reply:
e already contains "no executable for `clippy-driver` found in PATH", should I leave that as is and just drop the " component" from the second message?

Reply:
Yep!